### ISSUE TITLE
psmeca: Last few columns of input data should be optional

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#2016]
This issue was reported in https://github.com/GenericMappingTools/gmt/issues/661#issuecomment-552072249.

To make it more clear, here is an example.
```
gmt psmeca -R239/240/34/35.2 -JX8c -Baf -Sa2c << END > test.ps
# lon   lat    depth strike  dip  rake mag
239.384 34.556 12.   180     18   -88  5
END
```

For **-Sa**, the format of the input  data is:
```
lon  lat  depth  strike  dip  rake  mag  newX  newY  title
```

In GMT5, the last three columns are optional and the example above works. But in GMT6,  only the last column is optional, and the example fails with an error:
```
psmeca [ERROR]: Mismatch between actual (7) and expected (9) fields near line 2 in file <Standard Input>
```

psmeca in GMT5 works because it reads the input data via `scanf` statements. However, in GMT6, the psmeca I/O was rewritten (by @PaulWessel?) to use the standard GMT I/O mechanism. 

@PaulWessel Is it possible to make the last three columns optional?

**Comment by carltape** [comment]
Thanks @seisman. Another example is for the -Sm flag, which should require 10 columns (in GMT 5.4.4) but now requires the two optional columns (11 and 12). My error is this:
psmeca [ERROR]: Mismatch between actual (10) and expected (12) fields near line 1 in file beachballs_ipts1_iref1_lune_psmeca
Either a fix to the psmeca documentation (columns are NOT optional) or an update to the psmeca I/O would be needed.

**Comment by PaulWessel** [comment]
The decision on number of columns to expect is done apparently in a few places, incosistently.  In parsing -Sm we have

```
					case 'm':
						Ctrl->S.readmode = READ_TENSOR;	Ctrl->S.n_cols = 12;
						break;
```

while at the start of the program we find

```
	if (Ctrl->S.readmode == READ_CMT)
		last = 11;
	else if (Ctrl->S.readmode == READ_AKI)
		last = 7;
	else if (Ctrl->S.readmode == READ_PLANES)
		last = 8;
	else if (Ctrl->S.readmode == READ_AXIS)
		last = 13;
	else if (Ctrl->S.readmode == READ_TENSOR)
		last = 10;

	if (Ctrl->O2.active) Ctrl->S.n_cols--;	/* No depth */
```

So last seems to have the right number but we then call

	GMT_Set_Columns (API, GMT_IN, Ctrl->S.n_cols, GMT_COL_FIX);

so now we are back to 12 again.   Before I try to come up with a revision to this, could @seisman try to replace the GMT_COL_FIX arg with GMT_COL_VAR and see if the problem goes away first?

**Comment by seisman** [comment]
Yes, it works!

**Comment by PaulWessel** [comment]
My work here is done.

**Comment by seisman** [comment]
Nice fix! I didn't expect it so simple. I'll do more tests and open a PR later.

**Comment by PaulWessel** [comment]
When psmeca was written by Kurt Feigl and Guinevieve Patau in Paris back in the 1990s they took psxy and hacked it into psmeca.  Since then, psxy has undergone many changes but there is one similarity and that is psxy can accept the symbol code as part of the data record and that means the number of data columns vary.  When GMT5 rewrote the i/o I allowed for this via the GMT_COL_VAR mode which is only used in psxy/psxyz and also needed in psmeca.  Of course, the rapid porting of old modules to GMT5 would sometimes have side-effects, especially for modules that I don't use myself.  If there are similar situations in pscoupe etc then check if GMT_COL_VAR is needed that too.

**Comment by seisman** [comment]
The psmeca issue was fixed in #2018. Still need to do similar work to pscoupe.
